lib/src/Base/Optim/openturns/BonminProblem.hxx

```diff
@@ -179,7 +178,8 @@ public:
 private:
   const OptimizationProblem optimProblem_;
   const Point startingPoint_;
-  MemoizeFunction objectiveFunction_;
+  Sample evaluationInputHistory_;
```

Comment:
So if I understand well, the logic is to avoid the creation of MemoizeFunction in favor of algorithms histories? Is there a way to reuse the function history if the given objective function is a MemoizeFunction? For me, it is not clear if we have to automate things or to explain them in the documentation, but it can be a huge penalty to store everything in multiple places, both in terms of memory footprint and CPU performance

Reply:
yes, here we choose to avoid storring everything in a memoize